src/app/shared/util/metrics.util.ts

```diff
@@ -33,13 +60,22 @@ export const filterPodcastMetrics = (filter: FilterModel, podcastMetrics: Podcas
 
 export const filterEpisodeMetrics =
   (filter: FilterModel, episodeMetrics: EpisodeMetricsModel[], metricsType: MetricsType): EpisodeMetricsModel[] => {
-    if (filter.podcast && filter.episodes && filter.interval && episodeMetrics) {
-      const metricsProperty = filter.interval.key + metricsType.charAt(0).toUpperCase() + metricsType.slice(1);
-      return episodeMetrics.filter(metric => metric.seriesId === filter.podcast.seriesId &&
-      filter.episodes.map(e => e.id).indexOf(metric.id) !== -1 &&
-      metric[metricsProperty]);
-    }
-  };
+  if (filter.podcast && filter.episodes && filter.interval && filter.beginDate && filter.endDate && episodeMetrics) {
+    const metricsProperty = filter.interval.key + metricsType.charAt(0).toUpperCase() + metricsType.slice(1);
```

Comment:
could lift `filter.interval.key + metricsType.charAt(0).toUpperCase() + metricsType.slice(1)` into a small helper? since it's used in each of these. /shrug